src/components/inputs/DateTimePicker.tsx

```diff
@@ -873,22 +898,214 @@ interface DateTimePickerProps extends Omit<ReactDatePickerProps, 'onChange'> {
-	 *  	number: measure in px
-	 *  	string: any measure in CSS syntax
+	/**
+	 * Input width: <br/>
+	 *  <li>`fit`: shorthand for fit-content</li>
+	 *  <li>`fill`: semantic alternative for `100%`</li>
+	 *  <li>number: measure in px</li>
+	 *  <li>string: any measure in CSS syntax</li>
 	 */
 	width?: 'fit' | 'fill' | string | number;
-	/** Use Custom Component to trigger the picker */
+	/**
+	 * Use a custom component instead of the default one.
+	 * The component will be cloned by react-datepicker.
+	 * See "With Custom Input" section for more details.
+	 */
 	CustomComponent?: React.ComponentType<{
-		value?: string | undefined;
+		value?: string;
 		onClick?: React.ReactEventHandler;
 	}>;
 	/** Disable the input */
 	disabled?: boolean;
 }
 
+interface ReactDatePickerCustomInputProps
+	extends Pick<
+		InputHTMLAttributes<HTMLInputElement>,
+		| 'value'
+		| 'onBlur'
+		| 'onChange'
+		| 'onClick'
+		| 'onFocus'
+		| 'onKeyDown'
+		| 'id'
+		| 'name'
+		| 'form'
+		| 'autoFocus'
+		| 'placeholder'
+		| 'disabled'
+		| 'autoComplete'
+		| 'className'
+		| 'title'
+		| 'readOnly'
+		| 'required'
+		| 'tabIndex'
+		| 'aria-describedby'
+		| 'aria-invalid'
+		| 'aria-labelledby'
+		| 'aria-required'
+	> {
+	value?: string;
+}
+
+type DateTimePickerInputProps = Omit<InputProps, keyof ReactDatePickerCustomInputProps> & {
+	width: ContainerProps['width'];
+	isClearable: boolean;
+	onClear: IconButtonProps['onClick'];
+};
+
+type DateTimePickerChipInputProps = Omit<
+	ChipInputProps<Date>,
+	keyof ReactDatePickerCustomInputProps
+> & {
+	width: ContainerProps['width'];
+	chipValue: Date | null;
+	chipProps: Partial<ChipProps> | undefined;
+	/** Redefine onChange for ChipInput to avoid having it overwritten by react-datepicker */
+	handleChipChange: (items: DateChipItem[]) => void;
+};
+
+interface InputIconsProps {
+	showClear: boolean;
+	onClear: IconButtonProps['onClick'];
+	onClick: IconButtonProps['onClick'];
+}
+
+type DateChipItem = ChipItem<Date>;
+
+const buildInputIcons = ({
+	showClear,
+	onClear,
+	onClick
+}: InputIconsProps): NonNullable<InputProps['CustomIcon']> =>
+	function InputIcons({ hasError }): JSX.Element {
+		return (
+			<InputIconsContainer>
+				{showClear && (
+					<CustomIconButton
+						icon="CloseOutline"
+						size="large"
+						onClick={onClear}
+						backgroundColor="transparent"
+					/>
+				)}
+				<CustomIconButton
+					icon="CalendarOutline"
+					size="large"
+					backgroundColor="transparent"
+					onClick={onClick}
+					iconColor={hasError ? 'error' : 'text'}
+				/>
+			</InputIconsContainer>
+		);
+	};
+
+const DateTimePickerInput = React.forwardRef<
+	HTMLDivElement,
+	DateTimePickerInputProps & { [K in keyof ReactDatePickerCustomInputProps]: never }
+>(function DateTimePickerInputFn(
+	{
+		width,
+		onClear,
+		isClearable,
+		placeholder,
+		...rest
+	}: DateTimePickerInputProps & ReactDatePickerCustomInputProps,
+	ref
+) {
+	const { value, onClick = noop } = rest;
```

Comment:
My intention was to use the props inside the DateTimePickerInputFn to split only the props which are a 1:1 with the ones of the Input from the one which are not.

Value and onClick need to be passed to the Input, but I also need them inside the icons. Doing this way allows me to avoid having to repeat the props in the Input.
So instead of doing this
```tsx
<Input CustomIcon={InputIconsComponent} label={placeholder} value={value} onClick={onClick} {...rest} />
```
the `onClick` and the `value` are included inside the `rest`.

If it's not clear, I can explicit the two also for the Input